src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,66 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final DateTimeFormatter DATE_TIME_FORMATTER =
+            DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final String TITLE_FOR_REPORT = "Report for period ";
+    private static final String DASH_FOR_REPORT = " - ";
+    private static final String NEW_LINE = System.lineSeparator();
+    private static final String STRING_FOR_SPLIT = " ";
+    private static final int NUMBER_FOR_CORRECTLY_DATE = 1;
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int NUMBER_OF_HOURS_INDEX = 2;
+    private static final int SALARY_PER_HOUR = 3;
+
+    public String getSalaryInfo(String[] names, String[] data,
+                                String dateFrom, String dateTo) {
+        LocalDate fromDate = LocalDate.parse(dateFrom, DATE_TIME_FORMATTER);
+        LocalDate toDate = LocalDate.parse(dateTo, DATE_TIME_FORMATTER);
+        List<EmployeeWorkingDay> peoplesSalaries = parseEmployeeWorkingDay(data);
+        return getSalary(names, peoplesSalaries, fromDate, toDate);
+    }
+
+    private String getSalary(String[] names, List<EmployeeWorkingDay> peoplesSalaries,
+                             LocalDate fromDate, LocalDate toDate) {
+        StringBuilder report = new StringBuilder(TITLE_FOR_REPORT);
```

Comment:
Не вижу смысла иметь отдельный getSalary() метод. Он не выполняет какую-то отдельную операцию.
```suggestion
        List<EmployeeWorkingDay> peoplesSalaries = parseEmployeeWorkingDay(data);

        StringBuilder report = new StringBuilder(TITLE_FOR_REPORT);
```